### PR TITLE
Update Automated build documentation

### PR DESCRIPTION
**Pull request by apielmicroej**

### CODE REVIEW

**Comment by fpinel-microej** [comment]
line 113: There is no "Working Offline" section anymore, this should be rewritten

**Comment by fpinel-microej** [comment]
l161 : The central repository archive isn't named like this (anymore), it was named `central-repository-3.20.0.zip` when I downloaded it

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst [hunk 1]
```diff
@@ -318,46 +245,74 @@ In this example, we will create a very simple module using the Sandbox Applicati
...
+.. note::
+   For more details about MicroEJ Applications development, refer to the :ref:`Application Developer Guide <application-developer-guide>`.
+
+Upload to your Git repository
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note::
+   We need IP of the Docker Bridge Network, here we consider that it's ``172.17.0.1`` but you can check with the command ``ip addr show docker0`` on the docker host.
+
+#. Open the project directory, create a file name ``Jenkinsfile`` and copy this content inside:
```
Comment (fpinel-microej) [inline Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:257]:
```suggestion
#. Open the project directory, create a file named ``Jenkinsfile`` and copy this content inside:
```

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst [hunk 2]
```diff
@@ -318,46 +245,74 @@ In this example, we will create a very simple module using the Sandbox Applicati
...
+
+    .. literalinclude:: resources/Jenkinsfile
+        :language: groovy
+
+#. Create a directory named ``ivy``, create a file name ``ivysettings-artifactory.xml`` and copy this content inside:
```
Comment (fpinel-microej) [inline Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:262]:
```suggestion
#. Create a directory named ``ivy``, create a file named ``ivysettings-artifactory.xml`` and copy this content inside: 
```

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst [hunk 3]
```diff
@@ -318,46 +245,74 @@ In this example, we will create a very simple module using the Sandbox Applicati
...
+
+    .. literalinclude:: resources/Jenkinsfile
+        :language: groovy
+
+#. Create a directory named ``ivy``, create a file name ``ivysettings-artifactory.xml`` and copy this content inside: 
+
+    .. literalinclude:: resources/ivysettings-artifactory.xml
+        :language: xml
+
+This file configures MicroEJ Module Manager to import and publish modules from the Artifactory repositories described in this tutorial. Please refer to :ref:`mmm_settings_file` section for more details.
```
Comment (fpinel-microej) [inline Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:267]:
```suggestion
This file configures the MicroEJ Module Manager to import and publish modules from the Artifactory repositories described in this tutorial. Please refer to the :ref:`mmm_settings_file` section for more details.
```

**Comment by fpinel-microej** [comment]
On my end, build gets stuck at publish stage without fully completing. Logs:
```
Started by user [admin](http://localhost:8080/user/admin)
 > git rev-parse --resolve-git-dir /var/jenkins_home/caches/git-5b536b9cc249f73ea0bafeb5605b962e/.git # timeout=10
Setting origin to http://172.17.0.1:3000/fpinel/helloworld.git
 > git config remote.origin.url http://172.17.0.1:3000/fpinel/helloworld.git # timeout=10
Fetching origin...
Fetching upstream changes from origin
 > git --version # timeout=10
 > git --version # 'git version 2.39.2'
 > git config --get remote.origin.url # timeout=10
 > git fetch --tags --force --progress -- origin +refs/heads/*:refs/remotes/origin/* # timeout=10
Seen branch in repository origin/main
Seen 1 remote branch
Obtained Jenkinsfile from 882f2ad86431b31b94e7ba8827612d7ae7f01944
[Pipeline] Start of Pipeline
[Pipeline] node
Running on [Jenkins](http://localhost:8080/computer/(built-in)/) in /var/jenkins_home/workspace/Hello_World_main
[Pipeline] { ([hide](http://localhost:8080/job/Hello%20World/job/main/3/console#))
[Pipeline] stage
[Pipeline] { (Declarative: Checkout SCM)
[Pipeline] checkout
The recommended git tool is: NONE
No credentials specified
 > git rev-parse --resolve-git-dir /var/jenkins_home/workspace/Hello_World_main/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url http://172.17.0.1:3000/fpinel/helloworld.git # timeout=10
Fetching without tags
Fetching upstream changes from http://172.17.0.1:3000/fpinel/helloworld.git
 > git --version # timeout=10
 > git --version # 'git version 2.39.2'
 > git fetch --no-tags --force --progress -- http://172.17.0.1:3000/fpinel/helloworld.git +refs/heads/*:refs/remotes/origin/* # timeout=10
Checking out Revision 882f2ad86431b31b94e7ba8827612d7ae7f01944 (main)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 882f2ad86431b31b94e7ba8827612d7ae7f01944 # timeout=10
Commit message: "Add Hello World application"
 > git rev-list --no-walk 882f2ad86431b31b94e7ba8827612d7ae7f01944 # timeout=10
[Pipeline] }
[Pipeline] // stage
[Pipeline] withEnv
[Pipeline] {
[Pipeline] isUnix
[Pipeline] withEnv
[Pipeline] {
[Pipeline] sh
+ docker inspect -f . microej/sdk:5.8.1
.
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] withDockerContainer
Jenkins seems to be running inside container c94ec2ce8124f34eaff7c5822e42f2253cc35e53571b8f9f87da4ffcee192300
$ docker run -t -d -u 0:0 -e ACCEPT_MICROEJ_SDK_EULA_V3_1B=YES -w /var/jenkins_home/workspace/Hello_World_main --volumes-from c94ec2ce8124f34eaff7c5822e42f2253cc35e53571b8f9f87da4ffcee192300 -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** microej/sdk:5.8.1 cat
$ docker top 77d5036ae7c3e980cbe3b22d0eda017450fc4e06d49c02153f93296a5bed191f -eo pid,comm
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Publish)
[Pipeline] sh
+ mmm publish shared -r ivy/ivysettings-artifactory.xml
[easyant-configure-easyant] :: Apache Ivy 2.7.1-local-20230703110641 - 20230703110641 :: http://ant.apache.org/ivy/ ::
[easyant-configure-easyant] :: loading settings :: file = /opt/buildKit/microej-build-repository/ivysettings.xml
Loading System Plugins...
[easyant-load-module] Loading build module : /var/jenkins_home/workspace/Hello_World_main/module.ivy
     [echo] Building com.example hello-world with com.is2t.easyant.buildtypes#build-application...
[easyant-load-module] :: loading settings :: file = ivy/ivysettings-artifactory.xml

clean-std:clean:
     [echo] Cleaning /var/jenkins_home/workspace/Hello_World_main/target~...
   [delete] Deleting directory /var/jenkins_home/workspace/Hello_World_main/target~

-jazz-toolkit:fallback-nojazz:
Trying to override old definition of datatype junitLogPublisher

-build-application:init:
    [mkdir] Created dir: /var/jenkins_home/workspace/Hello_World_main/target~/ant
     [copy] Copying 3 files to /var/jenkins_home/workspace/Hello_World_main/target~/ant/lib
     [copy] Copying /opt/buildKit/microej-build-repository/org/apache/ant/ant/1.9.9/ant-1.9.9.jar to /var/jenkins_home/workspace/Hello_World_main/target~/ant/lib/ant-1.9.9.jar
     [copy] Copying /opt/buildKit/microej-build-repository/org/apache/ant/ant-launcher/1.9.9/ant-launcher-1.9.9.jar to /var/jenkins_home/workspace/Hello_World_main/target~/ant/lib/ant-launcher-1.9.9.jar
     [copy] Copying /opt/buildKit/microej-build-repository/org/apache/ant/ant-junit/1.8.4/ant-junit-1.8.4.jar to /var/jenkins_home/workspace/Hello_World_main/target~/ant/lib/ant-junit-1.8.4.jar

-ivy-provisioning:resolve:
[ivy:resolve] Configure microej module manager semantic cache for localRepository resolver.
[ivy:resolve] Configure microej module manager semantic cache for modulesSnapshotRepository resolver.
[ivy:resolve] Configure microej module manager semantic cache for modulesReleaseRepository resolver.
[ivy:resolve] Configure microej module manager semantic cache for microejModulesRepository resolver.
```

**Review by fpinel-microej — COMMENTED** [review]
see other comments

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst [hunk 4]
```diff
@@ -196,22 +162,36 @@ In this section, we will import MicroEJ repositories into Artifactory repositori
 #. Click :guilabel:`Upload`. At the end of upload, click on :guilabel:`Import`. Upload and import may take some time.
 
 Artifactory is now hosting all required MicroEJ modules. 
-Go to :guilabel:`Administration` > :guilabel:`Artifactory` > :guilabel:`Artifacts` and check that the repository ``microej-module-repository`` does contain modules as shown in the figure below.
+Go to :guilabel:`Application` > :guilabel:`Artifactory` > :guilabel:`Artifacts` and check that the repository ``microej-module-repository`` does contain modules as shown in the figure below.
 
 .. image:: images/tuto_microej_cli_artifactory_module_preview.PNG
```
[the remark this answers is not in the document]
Comment (apielmicroej) [reply Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:167]: I have updated the image.

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst [hunk 5]
```diff
@@ -131,43 +110,30 @@ It bundles Foundation Library APIs and numerous Add-On Libraries.
...
 #. Click on :guilabel:`Save`.
-#. Go to :guilabel:`Administration` > :guilabel:`Identity and Access` > :guilabel:`Permissions`.
+#. Go to :guilabel:`Administration` > :guilabel:`User Management` > :guilabel:`Permissions`.
 #. Click on :guilabel:`Anything` entry (do not check the line), then go to :guilabel:`Users` tab
-#. Click on :guilabel:`anonymous` and check :guilabel:`Deploy/Cache` permission in the :guilabel:`Repositories` category.
-#. Click on :guilabel:`Save and finish`.
+#. Click on :guilabel:`anonymous` and check :guilabel:`Deploy/Cache` permission in the :guilabel:`Selected Users Repositories` category.
+#. Click on :guilabel:`Save`.
 
 Next steps will involve uploading large files, so we have to augment the file upload maximum size accordingly:
```
Comment (fpinel-microej) [inline Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:134]:
```suggestion
Next steps will involve uploading large files, so we have to increase the file upload maximum size accordingly:
```

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst [hunk 6]
```diff
@@ -79,43 +77,24 @@ Artifactory will host MicroEJ modules in 3 repositories:
...
+    .. literalinclude:: resources/docker-compose.yaml
+        :language: yaml
 
-This file configures MicroEJ Module Manager to import and publish modules from the Artifactory repositories described in this tutorial. Please refer to :ref:`mmm_settings_file` section for more details.
+#. Create an other file name ``Dockerfile`` and copy this content: 
+
+    .. literalinclude:: resources/Dockerfile
+        :language: dockerfile
+
+#. In this directory launch the command ``docker compose up -d``. After a few moment you must have three running containers (named jenkins, gitea and artifactory).
```
[the remark this answers is not in the document]
Comment (apielmicroej) [reply Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst]: There already are two empty lines after this line.

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst [hunk 7]
```diff
@@ -79,43 +77,24 @@ Artifactory will host MicroEJ modules in 3 repositories:
...
+        :language: yaml
 
-This file configures MicroEJ Module Manager to import and publish modules from the Artifactory repositories described in this tutorial. Please refer to :ref:`mmm_settings_file` section for more details.
+#. Create another file named ``Dockerfile`` and copy this content: 
+
+    .. literalinclude:: resources/Dockerfile
+        :language: dockerfile
+
+#. In this directory, launch the command ``docker compose up -d``. After a few moments you should have three running containers (named jenkins, gitea and artifactory). 
+Using ``docker compose ps`` will show if containers started properly. Logs can be viewed with ``docker compose logs``.
```
Comment (fpinel-microej) [inline Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:96]:
```suggestion
#. In this directory, launch the command ``docker compose up -d``. After a few moments you should have three running containers (named jenkins, gitea and artifactory). 

Using ``docker compose ps`` will show if containers started properly. Logs can be viewed with ``docker compose logs``. 
```
Or merge both lines together, a bullet point should be one line and a line skip or else ReST breaks. tested in local build

**Review by fpinel-microej — APPROVED** [review]

**Pull request merged**